fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java

```diff
@@ -134,8 +134,16 @@ public boolean columnEqual(Expr srcExpr) {
             srcTableName = srcSlotRef.getTableName();
         }
         TableName thisTableName = tblName;
-        if (thisTableName == null && desc != null) {
-            thisTableName = getTableName();
+        if (desc != null && desc.getParent() != null && desc.getParent().getRef() != null) {
+            thisTableName = desc.getParent().getRef().getName();
+        } else if (!desc.getSourceExprs().isEmpty()) {
+            List<Expr> sourceExprs = desc.getSourceExprs();
+            Expr sourceExpr = sourceExprs.get(0);
```

Comment:
source exprs get (0) ? how about others?

Reply:
When slotref can't get db.table from "desc.getParent().getRef().getName", I think the sourceExpr of partitionExprs is only one slotRef. The union situation maybe happen? 
Or I check all sourceExpr's tablename, if them is the same, then it's the actual table name.